common/scala/src/main/scala/org/apache/openwhisk/common/Logging.scala

```diff
@@ -184,13 +185,71 @@ case class LogMarkerToken(component: String,
                           state: String,
                           subAction: Option[String] = None,
                           tags: Map[String, String] = Map.empty) {
+  private var finishToken: LogMarkerToken = _
+  private var errorToken: LogMarkerToken = _
+  private var _counter: KCounter = _
+  private var _histogram: KHistogram = _
```

Comment:
It would be safe to use `var` here as the variables would get initialized to "same" values. Kamon would ensure that metric instance for given name and tag are always same

Reply:
It'd be nice to add that consideration as a comment here, along the lines of:

```
// Thread-safety is not an issue here, because Kamon makes sure the instances (given the same key) are always the same, so a missed update is not harmful
```